ctapipe/irf/select.py

```diff
@@ -0,0 +1,205 @@
+import astropy.units as u
+import numpy as np
+from astropy.table import QTable, vstack
+from pyirf.simulations import SimulatedEventsInfo
+from pyirf.spectral import PowerLaw, calculate_event_weights
+from pyirf.utils import calculate_source_fov_offset, calculate_theta
+
+from ..core import Component, Provenance, QualityQuery
+from ..core.traits import List, Unicode
+from ..io import TableLoader
+
+
+class EventSelector(Component):
+    def __init__(self, event_pre_processor, kind, file, target_spectrum, **kwargs):
+        super().__init__(**kwargs)
+
+        self.epp = event_pre_processor
+        self.target_spectrum = target_spectrum
+        self.kind = kind
+        self.file = file
+
+    def load_preselected_events(self, chunk_size, obs_time, fov_bins):
+        opts = dict(load_dl2=True, load_simulated=True, load_dl1_parameters=False)
+        with TableLoader(self.file, **opts) as load:
+            Provenance().add_input_file(self.file)
```

Comment:
That should be in tableloader, so not needed.

Reply:
fixed